tests/test_categories.py

```diff
@@ -1,136 +1,147 @@
+    cc_client.update_categories("file", target=categories)
+    remote_categories = cc_client.list_categories("file")
+    assert set(categories).issubset(
+        remote_categories["path"]
+    ), "Remote categories do not match initial categories"
+
 
-def test_category_addition():
+def test_category_addition() -> None:
     """Test that new categories are added correctly (delete=False)."""
     cc_client = CashCtrlClient()
-    cc_client.update_categories('file', target=more_categories, delete=False)
-    remote_categories = cc_client.list_categories('file')
-    assert set(more_categories).issubset(remote_categories['path']), (
-        "Not all initial categories appear in remote categories")
+    cc_client.update_categories("file", target=more_categories, delete=False)
+    remote_categories = cc_client.list_categories("file")
+    assert set(more_categories).issubset(
+        remote_categories["path"]
+    ), "Not all initial categories appear in remote categories"
+
 
 @pytest.mark.skip(reason="One category on test account can not be deleted.")
-def test_category_deletion():
+def test_category_deletion() -> None:
     """Test that categories are deleted when specified."""
     cc_client = CashCtrlClient()
-    cc_client.update_categories('file', target=more_categories, delete=True)
-    remote_categories = cc_client.list_categories('file')
+    cc_client.update_categories("file", target=more_categories, delete=True)
+    remote_categories = cc_client.list_categories("file")
     target = pd.Series(more_categories)
-    assert all([target.str.startswith(node).any()
-                for node in remote_categories['path']]), (
-        "Remote categories do not match `more_categories'")
+    assert all(
+        [target.str.startswith(node).any() for node in remote_categories["path"]]
+    ), "Remote categories do not match `more_categories`"
+
 
 @pytest.mark.skip(reason="One category on test account can not be deleted.")
-def test_category_removal():
+def test_category_removal() -> None:
     """Test that all categories are deleted when specified."""
     cc_client = CashCtrlClient()
-    cc_client.update_categories('file', target=[], delete=True)
-    remote_categories = cc_client.list_categories('file')
+    cc_client.update_categories("file", target=[], delete=True)
+    remote_categories = cc_client.list_categories("file")
     assert len(remote_categories) == 0, "Some remote categories remain."
 
-def test_invalid_path():
+
+def test_invalid_path() -> None:
     """Test the system's response to invalid category paths."""
-    invalid_categories = ['not/a/valid/path', '']
+    invalid_categories = ["not/a/valid/path", ""]
     cc_client = CashCtrlClient()
-    with pytest.raises(Exception):
-        cc_client.update_categories('file', target=invalid_categories[[0]])
-    with pytest.raises(Exception):
-        cc_client.update_categories('file', target=invalid_categories[[1]])
```

Comment:
Did you drop this test on purpose?

Reply:
I've checked this tests and they were passing just because of this error:

```bash
    def test_invalid_path() -> None:
        """Test the system's response to invalid category paths."""
        invalid_categories = ["not/a/valid/path", ""]
        cc_client = CashCtrlClient()
>       target=invalid_categories[[1]]
E       TypeError: list indices must be integers or slices, not list
tests/test_categories.py:75: TypeError
====================================================================================== short test summary info ======================================================================================
FAILED tests/test_categories.py::test_invalid_path - TypeError: list indices must be integers or slices, not list
```

I've fixed first part when invalid path is passed, but when `""` is passed our function works fine

Reply:
Thanks!